src/silo/storage/position.cpp

```diff
@@ -0,0 +1,160 @@
+#include "silo/storage/position.h"
+
+#include <spdlog/spdlog.h>
+
+template <typename SymbolType>
+silo::Position<SymbolType> silo::Position<SymbolType>::fromInitiallyDeleted(
+   typename SymbolType::Symbol symbol
+) {
+   silo::Position<SymbolType> position;
+   position.symbol_whose_bitmap_is_deleted = symbol;
+   return position;
+}
+
+template <typename SymbolType>
+silo::Position<SymbolType> silo::Position<SymbolType>::fromInitiallyFlipped(
+   typename SymbolType::Symbol symbol
+) {
+   silo::Position<SymbolType> position;
+   position.symbol_whose_bitmap_is_flipped = symbol;
+   return position;
+}
+
+template <typename SymbolType>
+void silo::Position<SymbolType>::addValues(
+   typename SymbolType::Symbol symbol,
+   const std::vector<uint32_t>& values,
+   size_t current_offset,
+   size_t interval_size
+) {
+   if (symbol == symbol_whose_bitmap_is_deleted) {
+      return;
+   }
+   if (!values.empty()) {
+      bitmaps[symbol].addMany(values.size(), values.data());
+   }
+   if (symbol == symbol_whose_bitmap_is_flipped) {
+      bitmaps[symbol].flip(current_offset, current_offset + interval_size);
+   }
+}
+
+template <typename SymbolType>
+std::optional<typename SymbolType::Symbol> silo::Position<SymbolType>::getHighestCardinalitySymbol(
+   uint32_t sequence_count
+) {
+   if (symbol_whose_bitmap_is_deleted.has_value()) {
+      throw std::runtime_error(fmt::format(
+         "Symbol '{}' is currently deleted. Cannot restore it implicitly and cannot calculate its "
+         "cardinality as we do not have information about missing symbols",
+         SymbolType::symbolToChar(*symbol_whose_bitmap_is_deleted)
+      ));
+   }
+
+   std::optional<typename SymbolType::Symbol> max_symbol = std::nullopt;
+   uint32_t max_count = 0;
+
+   for (const auto& symbol : SymbolType::SYMBOLS) {
+      roaring::Roaring& bitmap = bitmaps[symbol];
+      bitmap.runOptimize();
+      bitmap.shrinkToFit();
```

Comment:
I don't know whether it's good to hide this side effect in a `getSomething` method.